### PR TITLE
multicall

### PR DESCRIPTION
**Pull request by rouzwelt**

### CODE REVIEW

File: src/multicall.rs [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+        multicall.add_call(dai_symbol_call);
+
+        let usdc = Address::from_hex("0x2791bca1f2de4661ed88a30c99a7a9449aa84174")?;
+        let usdc_symbol_call = MulticallCallItem {
+            address: usdc,
+            call: symbolCall {},
+        };
+        multicall.add_call(usdc_symbol_call);
+
+        let provider = ReadableClient::new_from_url("https://rpc.ankr.com/polygon".to_owned())?;
```
Comment (thedavidmeister) [inline src/multicall.rs:180]: no hardcoded rpc urls

File: src/multicall.rs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+}
+
+impl<T: SolCall> Multicall<T> {
+    /// adds a single call to the list of multicall calls
+    pub fn add_call(&mut self, call: MulticallCallItem<T>) {
+        self.calls.push(call);
+    }
+
+    /// clears the calls list
+    pub fn clear_calls(&mut self) {
```
Comment (thedavidmeister) [inline src/multicall.rs:108]: not tested

File: src/multicall.rs [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    pub call: T,
+}
+
+/// A struct that makes making multicall reads of same call types easy
+#[derive(Debug, Clone)]
+pub struct Multicall<T: SolCall> {
+    pub calls: Vec<MulticallCallItem<T>>,
+}
+
+impl<T: SolCall> Default for Multicall<T> {
```
Comment (thedavidmeister) [inline src/multicall.rs:95]: can this be derived?

File: src/multicall.rs [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+/// Multicall3 contract address on all supported chains
+/// safe to say Multicall3 is deployed at this address on all major evm chains,
+/// except a few handful of chains, notably zkSyncEra, Tron, and few testnets
+/// see: https://www.multicall3.com/deployments
+pub const MULTICALL3_ADDRESS: &str = "0xcA11bde05977b3631167028862bE2a173976CA11";
+
+sol! {
+    /// IMulticall3 contract interface
+    /// see: https://www.multicall3.com/abi
+    interface IMulticall3 {
```
Comment (thedavidmeister) [inline src/multicall.rs:16]: is there a way to get this from a sol dep?

**Review by thedavidmeister — CHANGES_REQUESTED** [review]

File: src/multicall.rs (hunk 3 above)
Comment (rouzwelt) [reply src/multicall.rs:95]: no, because of Solcall

**Pull request merged**